### PR TITLE
LPS-150036 [Raylife AP] Create Application Products Chart

### PR DESCRIPTION
**Pull request by joanagrego**

### CODE REVIEW

**Comment by joanagrego** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+								_defaultTitleFormat,
+								_defaultValueFormat,
+								_color
+							) => {
+								const title = Liferay.Language.get(data[0].id);
+								const percent = (data[0].ratio * 100).toFixed(
+									1
+								);
+
+								return `<div class="d-flex applications-products-tooltip p-2"'><spam class="d-flex w-100 mr-2">${title}</spam> <spam>${percent}%</spam></div>`;
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.js:134]: Please, remove it `'`.

```javascript
return `<div class="d-flex applications-products-tooltip p-2"><spam class="d-flex w-100 mr-2">${title}</spam> <spam>${percent}%</spam></div>`;
```

**Review by silasbsi — CHANGES_REQUESTED** [review]
Looks good to me, but needs some adjustments.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.css [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	background-color: #4c84ff;
+	content: '';
+	display: block;
+	height: 1px;
+	margin-top: 6px;
+	position: absolute;
+	width: 19px;
+}
+
+.applications-products-tooltip {
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.css:27]: Sure!

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+				const [productQuotesResult, applicationsResult] = results;
+
+				const columnsArr = [];
+				const colorsObj = {};
+
+				titleUpdate(applicationsResult?.value?.data?.totalCount, [
+					'h2',
+				]);
+
+				productQuotesResult?.value?.data?.items?.map((item, index) => {
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.js:51]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+						(application) =>
+							Number(application.productQuote) === item.productId
+					).length;
+
+					const shortDescription = Object.values(
+						item.shortDescription
+					)[0];
+					const [fullName] = Object.values(item.name);
+
+					let [productName] = Object.values(item.name);
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.js:62]: You're right!

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+					const countApplications = applicationsResult?.value?.data?.items?.filter(
+						(application) =>
+							Number(application.productQuote) === item.productId
+					).length;
+
+					const shortDescription = Object.values(
+						item.shortDescription
+					)[0];
+					const [fullName] = Object.values(item.name);
+
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.js:61]: ok

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.js (hunk 1 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.js:134]: done

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.css [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+	background-color: #fff;
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/src/agent-portal/fragments/applications-products/index.css:28]: Yeah! bg-neutral-0 on dialect. I'll change it.

**Comment by silasbsi** [comment]
LGTM :fire:

**Review by silasbsi — APPROVED** [review]

**Comment by silasbsi** [comment]
ci:test:sf

**Comment by silasbsi** [comment]
ci:forward

**Pull request closed without merging**